src/common/options/rgw.yaml.in

```diff
@@ -3999,3 +3999,13 @@ options:
   services:
   - rgw
   with_legacy: true
+- name: rgw_kafka_message_timeout
+  type: uint 
+  level: advanced
+  desc: This is the maximum time in milliseconds to deliver a message (including retries)
+  long_desc: Delivery error occurs when either the retry count or the message timeout are exceeded.
+    If set to zero, message will time out only based on retries.
```

Comment:
but we are not setting retries here, so if someone sets to zero what will happen?
since we are allowing setting timeout, we can also set max_retry? `message.send.max.retries` & `retry.backoff.ms`

Reply:
you are right, the retry default is too high.
will add as a conf parameter with a lower value.

the retry backoff is 100ms up to max of 1s. these values make sense. do you think we should expose these?

Reply:
100ms seems reasonable

Reply:
looking at the conf of the librdkafka version that we use: https://github.com/confluentinc/librdkafka/blob/v0.11.6/CONFIGURATION.md

the default retries are 2.
the backoof is 100ms
there is no max backoff (not sure there is exponential backoff).
anyway, until we upgrade our library version, we don't need to add these

Reply:
> looking at the conf of the librdkafka version that we use: https://github.com/confluentinc/librdkafka/blob/v0.11.6/CONFIGURATION.md
> 
> the default retries are 2. the backoof is 100ms there is no max backoff (not sure there is exponential backoff). anyway, until we upgrade our library version, we don't need to add these

sorry, but why not add it ? i get its value is 2 max_retry but since we are doing the PR for TTL, we can add max_retry config as well,  its only a single line change to add this config ? (purely for being a good citizen, since we add ttl, we also add the max_retry and let the default be same 2 as kafka) ?

Reply:
yes, will add. they changed it to MAX_INT in v1.6.1 (the version in rhel/centos9)

Reply:
only thing i'm not going to add is the max backoff - since it does not exist in the librdkafka versions we use

Reply:
could not get the max retry conf value to work.
at least for librdkafka v1.6.1, when message timeout was set to zero, the message never expired, even if max retry and the backoff numbers were set to low values.
so, to prevent this case, timeout cannot be set to zro, and will be set to 1ms if the user set it to zero.